sdk/internal/errorinfo/errorinfo.go

```diff
@@ -14,3 +14,24 @@ type NonRetriable interface {
 	error
 	NonRetriable()
 }
+
+// NonRetriableError marks the specified error as non-retriable.
+func NonRetriableError(err error) error {
+	return &nonRetriableError{err}
```

Comment:
It doesn't seem like it needs to be a pointer, btw.

Reply:
By convention we return errors as pointer-to-type. Plus, the receivers on `nonRetriableError` are by pointer.

Reply:
Discussed with Richard offline. Resolving comment